### PR TITLE
Added ev and evj command

### PR DESCRIPTION
**Pull request by yetmorecode**
Added ev and evj command:

```
user@linuxdev radare2-dev $ ./binr/radare2/radare2 --
 -- Find hexpairs with '/x a0 cc 33'
[0x00000000]> ev dbg.bta
dbg.btalgo = fuzzy ; Select backtrace algorithm [default, fuzzy, anal]
[0x00000000]> e dbg.btalgo=?
default
fuzzy
anal
[0x00000000]> evj dbg.bt
[{"name":"dbg.btalgo","value":"fuzzy","type":"str","desc":"Select backtrace algorithm","ro":false,"options":["default","fuzzy","anal"]},{"name":"dbg.btdepth","value":128,"type":"int","desc":"Depth of backtrace","ro":false}]
[0x00000000]> e?
|Usage: e [var[=value]]Evaluable vars
| e?asm.bytes     show description
| e??             list config vars with description
| e|ej            list config vars (in JSON)
| e-              reset config vars
| e*              dump config vars in r commands
| e!a             invert the boolean value of 'a' var
| eevar           open editor to change the value of var
| er [key]        set config key as readonly. no way back
| ec [k] [color]  set color for given key (prompt, offset, ...)
| et [key]        show type of given config variable
| ev|evj [key]    list config vars in verbose format (in JSON)
| e a             get value of var 'a'
| e a=b           set var 'a' the 'b' value
| e var=?         print all valid values of var
| env [k[=v]]     get/set environment variable
[0x00000000]>
```

Also added `RList *options` to RConfigNode and reworked dbg.btalgo in cconfig.c.

If this gets merged I can also follow up with reworking the rest of cconfig!

### CODE REVIEW

File: libr/core/cconfig.c [hunk 1]
```diff
@@ -2005,7 +2014,11 @@ R_API int r_core_config_init(RCore *core) {
 
 	SETPREF("dbg.bpinmaps", "true", "Force breakpoints to be inside a valid map");
 	SETCB("dbg.forks", "false", &cb_dbg_forks, "Stop execution if fork() is done (see dbg.threads)");
-	SETCB("dbg.btalgo", "fuzzy", &cb_dbg_btalgo, "Select backtrace algorithm");
+	n = NODECB("dbg.btalgo", "fuzzy", &cb_dbg_btalgo);
+	SETDESC(n, "Select backtrace algorithm");
+	SETOPTION(n, "default");
```
Comment (radare) [inline libr/core/cconfig.c]: what about using vararg here like this: `SETOPTIONS(n, "default, "fuzzy", "anal", NULL)`

File: libr/core/cmd_eval.c [hunk 2]
```diff
@@ -362,14 +379,15 @@ static int cmd_eval(void *data, const char *input) {
 			"Usage:", "e [var[=value]]", "Evaluable vars",
 			"e","?asm.bytes", "show description",
 			"e", "??", "list config vars with description",
-			"e", "", "list config vars",
+			"e|ej", "", "list config vars (in JSON)",
```
Comment (radare) [inline libr/core/cmd_eval.c]: this can confuse the recursive help command which is based on parsing the help

File: libr/core/cmd_eval.c [hunk 3]
```diff
@@ -214,6 +214,24 @@ static int cmd_eval(void *data, const char *input) {
 	case 'j': // json
 		r_config_list (core->config, NULL, 'j');
 		break;
+	case 'v': // verbose
+		switch (input[1]) {
+		default:
```
Comment (radare) [inline libr/core/cmd_eval.c]: just call r_config_list once and make this code just specify the 2nd arg value

Comment (radare) [inline libr/core/cmd_eval.c]: default should be always the last block in a switch case

File: libr/core/cconfig.c (hunk 1 above)
Comment (yetmorecode) [inline libr/core/cconfig.c]: Yes, thought about it too, would be prettier. Will do

File: libr/config/config.c [hunk 4]
```diff
@@ -83,39 +179,33 @@ R_API void r_config_list(RConfig *cfg, const char *str, int rad) {
...
-							char *sval = r_str_escape (val);
-							cfg->cb_printf ("\"%s\":\"%s\"", node->name, sval);
-							free (sval);
-						}
-					} else {
-						cfg->cb_printf ("\"%s\":\"%s\"",
-							node->name, val);
-					}
-					if (iter->n) {
+					if (!json_is_first) {
```
Comment (radare) [inline libr/config/config.c]: dont use negated conditionals if possible. it reads better to focus on positive conditionals

File: libr/config/config.c [hunk 5]
```diff
@@ -83,39 +179,33 @@ R_API void r_config_list(RConfig *cfg, const char *str, int rad) {
 			cfg->cb_printf ("%s\n", node->name);
 		}
 		break;
+	case 'J':
+		verbose = true;
 	case 'j':
-		cfg->cb_printf ("{");
+		json = true;
+		json_is_first = true;
```
Comment (radare) [inline libr/config/config.c]: last coding style rules says that static and locals should use camelcase, only public/exposed stuff uses the snake style

File: libr/config/config.c [hunk 6]
```diff
@@ -83,39 +179,33 @@ R_API void r_config_list(RConfig *cfg, const char *str, int rad) {
 			cfg->cb_printf ("%s\n", node->name);
 		}
 		break;
+	case 'J':
+		verbose = true;
```
Comment (radare) [inline libr/config/config.c]: add comments /* fall through */ or coverity will complain about missing break statement

**Comment by yetmorecode** [comment]
Thx for all the feedback :D I will re-iterate.

File: libr/config/config.c [hunk 7]
```diff
@@ -60,11 +155,12 @@ R_API void r_config_list(RConfig *cfg, const char *str, int rad) {
 		pfx = "\"e ";
 		sfx = "\"";
 	/* fallthrou */
+	case 'v':
+		verbose = true;
```
Comment (radare) [inline libr/config/config.c]: same here with the comment

**Comment by radare** [comment]
cool!, the pr looks good so far

**Comment by yetmorecode** [comment]
Ok, I pushed another commit. I hope its better now.

**Comment by radare** [comment]
lgtm, lets wait for travis. but seems like there's 1 broken test

screen shot 2017-04-27 at 21 17 44

**Comment by yetmorecode** [comment]
I might have an idea what this is. I will have a look.

**Comment by yetmorecode** [comment]
Ok, travis is green now.

**Comment by radare** [comment]
magic

**Pull request merged**